src/components/button/button.component.tsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+
+import { ButtonStyled } from './button.style'
+
+interface ButtonProps {
+  children: React.ReactNode
+  onClick?: () => void
```

Comment:
for sake of readability, we prefer to place "onSomething" as last props

so more like this:

```
interface Props {
   someProp: string
   someOtherProps: boolean
 
   onClick?: () => void
   onMouseMove?: () => void
}
```

thats gives you a clear separation between "input" and "output" props for component